clusterman/kubernetes/kubernetes_cluster_connector.py

```diff
@@ -49,28 +92,28 @@ class KubernetesClusterConnector(ClusterConnector):
 
     def __init__(self, cluster: str, pool: Optional[str]) -> None:
         super().__init__(cluster, pool)
-        self.kubeconfig_path = f'clusters.{cluster}.kubeconfig_path'
+        self.kubeconfig_path = staticconf.read_string(f'clusters.{cluster}.kubeconfig_path')
         self._safe_to_evict_annotation = staticconf.read_string(
             f'clusters.{cluster}.pod_safe_to_evict_annotation',
             default='cluster-autoscaler.kubernetes.io/safe-to-evict',
         )
 
     def reload_state(self) -> None:
         logger.info('Reloading nodes')
-        try:
-            kubernetes.config.load_kube_config(staticconf.read_string(f'{self.kubeconfig_path}'))
-        except TypeError:
-            error_msg = 'Could not load KUBECONFIG; is this running on Kubernetes master?'
-            if 'yelpcorp' in socket.getfqdn():
-                error_msg += '\nHint: try using the clusterman-k8s-<clustername> wrapper script!'
-            logger.error(error_msg)
-            raise
 
-        self._core_api = kubernetes.client.CoreV1Api()
+        self._core_api = CachedCoreV1Api(self.kubeconfig_path)
         self._pods = self._get_all_pods()
         self._nodes_by_ip = self._get_nodes_by_ip()
         self._pods_by_ip = self._get_pods_by_ip()
 
+    def get_removed_nodes_since_last_reload(self) -> Set[KubernetesNode]:
+        previous_nodes = self._nodes_by_ip
+        current_nodes = self._get_nodes_by_ip()
+
+        # todo jmalt: make sure this gives the desired results
```

Comment:
I think this todo still needs addressing?  since we're literally just comparing the length above, you _could_ just calculate the length of the difference here, or possibly reduce this to a list of IPs?  I think it's also fine the way it is, up to you.

Reply:
I think I'll keep it as-is, but this does raise a question of what happens if we lose and gain nodes in between autoscaler runs (e.g. some instances die and AWS replaces them). With this scenario, we'd consider that a capacity loss, but maybe we don't want to? If we don't, then maybe something like `len(prev_nodes) - len(current_nodes)` would be better.

What do you think?

Reply:
As for making sure that this functionality actually works end-to-end: the cluster connector is mocked in the itests so it's tricky to test there; I think I'll need to test it on dev somewhere (the infrastage Flink pool would do fine).